src/Tokenizer/TokensAnalyzer.php

```diff
@@ -633,6 +633,28 @@ public function isWhilePartOfDoWhile(int $index): bool
         return $tokens[$beforeStartIndex]->isGivenKind(T_DO);
     }
 
+    public function isEnumCase(int $caseIndex): bool
```

Comment:
Non-blocking, but consider adding proper `@throws LogicException when provided index does not point to token containing T_CASE`.

Reply:
Added